custom_components/easee/switch.py

```diff
@@ -24,26 +26,36 @@ class ChargerSwitch(ChargerEntity, SwitchEntity):
     async def async_turn_on(self, **kwargs):  # pylint: disable=unused-argument
         """Turn on the switch."""
         _LOGGER.debug("%s Switch turn on" % self._entity_name)
-        self.set_value_from_key(self._state_key, True)
-        self._state = True
-        self.async_write_ha_state()
         function_call = getattr(self.data.product, self._switch_func)
         try:
             await function_call(True)
+        except ForbiddenServiceException:
+            raise HomeAssistantError(
+                f"Forbidden {self._entity_name} turn_on - No access right"
+            ) from None
         except Exception:
             _LOGGER.error("Got server error while calling %s", self._switch_func)
+            return
+        self.set_value_from_key(self._state_key, True)
+        self._state = True
+        self.async_write_ha_state()
 
     async def async_turn_off(self, **kwargs):  # pylint: disable=unused-argument
         """Turn off the switch."""
         _LOGGER.debug("%s Switch turn off" % self._entity_name)
-        self.set_value_from_key(self._state_key, False)
-        self._state = False
-        self.async_write_ha_state()
         function_call = getattr(self.data.product, self._switch_func)
         try:
             await function_call(False)
+        except ForbiddenServiceException:
+            raise HomeAssistantError(
+                f"Forbidden {self._entity_name} turn_off - No access right"
+            ) from None
```

Comment:
And "return" here too.

Reply:
The `return` is never executed. The `raise HomeAssistantError` has some side-effect that I can't master. I  will change to a `_LOGGER.error`. Then the switch jumps back after 1-2 seconds, an error is logged, but no toast popup message.

Reply:
Right, I missed that it was raising a new exception. But it is strange then that it did not revert back to old value, since it should have not set the new value as it was...? Puzzling...